### ISSUE TITLE
How to use the UPLOAD_VAT_INVOICE feed?

### ISSUE THREAD

**Issue by sauladam** [amzn/selling-partner-api-docs#3644]
I've been trying for quite a while now and I just can't get the `UPLOAD_VAT_INVOICE` feed working.

All of our other feeds (`POST_ORDER_ACKNOWLEDGEMENT_DATA`, `POST_ORDER_FULFILLMENT_DATA`) do work just fine , but `UPLOAD_VAT_INVOICE` just won't.

any `POST` request to `/feeds/2021-06-30/feeds` with a `feed_type` of `UPLOAD_VAT_INVOICE` returns a `403 Unauthorized` with the message "Access to the resource is forbidden"

Yes, *Tax Invoicing* is ticked for the app roles:

![image](https://github.com/amzn/selling-partner-api-docs/assets/5918276/ef10aaa9-9b26-4ba7-b9ea-d500355778ce)

Most of the the issues with the same problem (like #404) refer to the feeds API version 2020-09-04 which still involves encryption of the attached feed documents. The encryption part was removed in version 2021-06-30 and the *createFeedDocument* operation `/feeds/2021-06-30/documents` does not return any encryption details anymore, so I'm guessing this is definitely not the issue.

Apart from that, we just follow the the docs like with the other feeds:

- create a feed document (works fine)
- write content to that document (in this case as `application/pdf`) (works fine)
- pass the document id to the createFeed operation (**fails**)

The docs even have a code example specifically for `UPLOAD_VAT_INVOICE` in Java: https://developer-docs.amazon.com/sp-api/docs/option-3-upload-invoices-through-sp-api-or-seller-central

We do follow the same steps. It still does not work. But the example seems quite outdated anyway. It mainly focuses on constructing the feed data, and then sets an "MWS auth token" (`request.setMWSAuthToken(configuration.getAmazon().getMwsAuthToken());`). So that's not really helpful.

So, just a sanity check - **does anybody have it working?** 
If you have it working - what are you doing differently than with other feeds? Do we need an RDT for this operation or something ([according to the docs we don't](https://developer-docs.amazon.com/sp-api/docs/tax-invoicing-restricted-role#apis))? 
There are a few issue disucssions claiming that it works, but others still seem unresolved (like #2301). 

We had it working fine on MWS (and it's still doing the job there until this last issue is resolved and we are finally able to migrate).

**Comment by sauladam** [comment]
Ok, I AKSHUALLY got it working - by rotating all credentials FOR THE THIRD TIME! 🥳 So if you run into the same problem, just don't give up!

**Comment by FabianKremer** [comment]
Hey @sauladam how did you solve the Problem ? Just by rotating the credentials ?

**Comment by sauladam** [comment]
> Hey @sauladam how did you solve the Problem ? Just by rotating the credentials ?

Pretty much, yes. I can't say 10000% for sure whether it wasn't some weird client side issue (like stale cache or something), but re-authorising the app (self authorisation) in the Developer Central and then using the new refresh token and secret somehow did the trick.

If you're still having issues, you might want to nuke your client side cache copletely just be be sure, and also triple check if you have the Tax Invoicing role set on your app as well as on your Developer Profile (https://sellercentral.amazon.de/developer/register). 

Other than that, there is really no further magic to this feed type and it works just like the other ones. No RDT, no encryption. Feel free to share some code if you're still having trouble 👍

**Comment by FabianKremer** [comment]
@sauladam Thank you for your answer, I will try rotating our credentials. 

You said no RDT,  so you don´t need the RDT for this feed? Becuase I thougt you need this if you're handling PII.

**Comment by sauladam** [comment]
@FabianKremer 

> You said no RDT, so you don´t need the RDT for this feed? Becuase I thougt you need this if you're handling PII.

No, you don't need an RDT for this. Here are the routes related to the Tax Invoicing role: https://developer-docs.amazon.com/sp-api/docs/tax-invoicing-restricted-role - the sensitive ones are marked with "RDT". 

In case of a self generated invoice, you already own the PII and hand it to Amazon yourself, so there is no point in a restricted token. The presigned document URLs you upload them to are already restricted for their specific use cases, so no need for encryption (or Amazon do it themselves). That's at least my understanding of the whole AWS infrastructure. 

Long story short: No RDT, no encryption :)

**Comment by Philomath88** [comment]
Has anyone encountered this error before when trying to upload an invoice? 

original-record-number  sku     error-code      error-type      error-message
1               79555   Error   The uploaded document is invalid.

**Comment by sauladam** [comment]
@Philomath88 Yeah I think I've seen something like this in the early tests. Can't remember what the issue was though. 

Are you setting the `application/pdf` content type on the document? Are you encoding it somehow (which you shouldn't)?
